Package/Image/ImageResolver.cs

```diff
@@ -0,0 +1,280 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+
+namespace OpenTap.Package
+{ 
+    /// <summary>
+    /// Resolves packages dependencies for an image.This should be able to resolve any set of package dependencies, but
+    /// it may take a long time in some edge cases. In most cases however it seems to settle quite quickly.
+    /// </summary>
+    class ImageResolver
+    {
+        CancellationToken cancelToken;
+        public ImageResolver(CancellationToken cancelToken)
+        {
+            this.cancelToken = cancelToken;
+        }
+           
+
+        public long Iterations;
+
+        public ImageResolution ResolveImage(ImageSpecifier image, PackageDependencyGraph graph)
```

Comment:
Consider splitting this method up in multiple methods. 220 lines in one method is a bit long 😄

Reply:
I actually don't think it would help much. The order of all these things are quite important.